tests/backup/backup_helper.go

```diff
@@ -5559,3 +5561,172 @@ func UpgradeKubevirt(versionToUpgrade string, workloadUpgrade bool) error {
+func CreateNestedDirectoriesWithFilesInPod(podName, podNamespace, basePath string, depth, levels, filesPerDirectory int) error {
+	var errChan = make(chan error)
+	for i := 0; i < depth; i++ {
+		dirPath := filepath.Join(basePath, fmt.Sprintf("level_%d", i))
+		err := CreateDirectoryInPod(podName, podNamespace, dirPath)
+		if err != nil {
+			fmt.Printf("Error creating directory %s: %s\n", dirPath, err.Error())
+			continue
+		}
+		if levels > 1 {
+			wg.Add(1)
+			go func(dirPath string) {
+				defer wg.Done()
+				errChan <- CreateDirectoryStructureInPod(podName, podNamespace, dirPath, depth, levels-1, filesPerDirectory)
+			}(dirPath)
+		}
+
+	}
+	if filesPerDirectory != 0 {
+		err := CreateFilesInPodDirectory(podName, podNamespace, basePath, filesPerDirectory)
+		if err != nil {
+			return fmt.Errorf("error creating files in %s: %s", basePath, err.Error())
+		}
+	}
+	go func() {
+		wg.Wait()
+		close(errChan)
+	}()
+
+	for err := range errChan {
+		if err != nil {
+			return fmt.Errorf("error creating nested directories: %s", err.Error())
+		}
+	}
+
+	return nil
+}
+
+// CreateDirectoryStructureInPod creates a directory structure with multiple levels and files per directory within a specified Pod.
+func CreateDirectoryStructureInPod(podName, podNamespace, basePath string, depth, levels, filesPerDirectory int) error {
+	for i := 0; i < depth; i++ {
+		dirPath := filepath.Join(basePath, fmt.Sprintf("level_%d", i))
+		err := CreateDirectoryInPod(podName, podNamespace, dirPath)
+		if err != nil {
+			fmt.Printf("Error creating directory %s: %s\n", dirPath, err.Error())
+			continue
+		}
+
+		if levels > 1 {
+			err := CreateDirectoryStructureInPod(podName, podNamespace, dirPath, depth, levels-1, filesPerDirectory)
+			if err != nil {
+				return fmt.Errorf("error creating nested directories: %s", err.Error())
+			}
+		} else {
+			err := CreateFilesInPodDirectory(podName, podNamespace, dirPath, filesPerDirectory)
+			if err != nil {
+				return fmt.Errorf("error creating files in %s: %s", dirPath, err.Error())
+			}
+		}
+	}
+	return nil
+}
+
+// CreateDirectoryInPod creates a directory within a specified Pod.
+func CreateDirectoryInPod(podName, podNamespace, dirPath string) error {
+	cmd := fmt.Sprintf("mkdir -p %s ", dirPath)
+	cmdArgs := []string{fmt.Sprintf("--kubeconfig=%v", CurrentClusterConfigPath), "exec", "-it", podName, "-n", podNamespace, "--", "/bin/sh", "-c", fmt.Sprintf("%s", cmd)}
+	_, err := kubectlExec(cmdArgs)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateFilesInPodDirectory creates a specified number of files per directory level in a Pod.
+func CreateFilesInPodDirectory(podName, podNameSpace, dirPath string, filesPerLevel int) error {
+	for i := 1; i <= filesPerLevel; i++ {
+		filePath := filepath.Join(dirPath, fmt.Sprintf("file%d.txt", i))
+		content := fmt.Sprintf("This is file %d in directory %s", i, dirPath)
```

Comment:
Can we pass another parameter called file size and create a file with the required size using `dd` command? This will be helpful if we want large volumes.